package/MDAnalysis/coordinates/XDR.py

```diff
@@ -179,53 +185,67 @@ def close(self):
 
     def _load_offsets(self):
         """load frame offsets from file, reread them from the trajectory if that
-        fails"""
+        fails. To prevent the competition of generating the same offset file
+        from multiple processes, an `InterProcessLock` is used."""
         fname = offsets_filename(self.filename)
+        lock_name = offsets_filename(self.filename,
+                                     ending='lock')
 
-        if not isfile(fname):
-            self._read_offsets(store=True)
-            return
-
-        # if offsets file read correctly, data will be a dictionary of offsets
-        # if not, data will be False
-        # if False, offsets should be read from the trajectory
-        # this warning can be avoided by loading Universe like:
-        # u = mda.Universe(data.TPR, data.XTC, refresh_offsets=True)
-        # refer to Issue #1893
-        data = read_numpy_offsets(fname)
-        if not data:
-            warnings.warn("Reading offsets from {} failed, "
-                          "reading offsets from trajectory instead\n"
-                          "Consider setting 'refresh_offsets=True' "
-                          "when loading your Universe".format(fname))
-            self._read_offsets(store=True)
-            return
-
-        ctime_ok = size_ok = n_atoms_ok = False
-
+        #  check if the location of the lock is writable.
         try:
-            ctime_ok = getctime(self.filename) == data['ctime']
-            size_ok = getsize(self.filename) == data['size']
-            n_atoms_ok = self._xdr.n_atoms == data['n_atoms']
-        except KeyError:
-            # we tripped over some old offset formated file
-            pass
-
-        if not (ctime_ok and size_ok and n_atoms_ok):
-            warnings.warn("Reload offsets from trajectory\n "
-                          "ctime or size or n_atoms did not match")
-            self._read_offsets(store=True)
-        else:
-            self._xdr.set_offsets(data['offsets'])
+            filelock = fasteners.InterProcessLock(lock_name)
```

Comment:
This part of the code looks rather ugly for the condition when the lock file cannot be created. Looking for advice on what we should do. For now, I am just creating a lock file inside `tmp` folder.